lib/livebook/session/data.ex

```diff
@@ -548,11 +559,14 @@ defmodule Livebook.Session.Data do
   defp queue_cell_evaluation(data_actions, cell, section) do
     data_actions
     |> update_section_info!(section.id, fn section ->
-      %{section | evaluation_queue: section.evaluation_queue ++ [cell.id]}
+      %{section | evaluation_queue: append_new(section.evaluation_queue, cell.id)}
```

Comment:
Appending is a relative expensive operation but I guess it is fine if we are keeping this within at most 100. Otherwise the :queue module may be a better fit.

Reply:
Yeah, I think in this case we are fine.